semigroups.cabal

```diff
@@ -128,6 +136,9 @@ library
     if flag(bytestring)
       build-depends: bytestring >= 0.9 && < 1
 
+    if flag(bytestring-builder) && impl(ghc < 7.8)
```

Comment:
@RyanGlScott if it's possible, it *will* happen. Also, matrix.h.h.o may test such configurations more aggressively in future to detect inaccurate version bounds. Version constraints on `ghc` shouldn't be used as proxy for package versions, as we're working towards breaking up correlations between boot-library versions and GHC's version. Same goes for conditionals on `__GLASGOW_HASKELL__` rather than `MIN_VERSION_...`. Those technical debts will break in future more likely.